src/components/Link/ListItem.js

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import classNames from "classnames";
+
+import colors from "../../theme/colors";
+import constants from "../../theme/constants";
+import { LinkListConsumer } from "./Context";
+import spacing from "../../theme/spacing";
+import typography from "../../theme/typography";
+import composeEventHandlers from "../../utils/composeEventHandlers";
+
+const ItemContainer = styled.div.attrs({
+  className: "list-item"
+})`
+  padding: 0 12px;
+  height: 36px;
+  display: flex;
+  color: ${colors.blackPearl};
+  align-items: center;
+  font-size: ${typography.size.kilo};
+  font-weight: ${typography.weight.regular};
+  border-radius: ${constants.borderRadius.small};
+
+  &.list-item--selected {
+    background-color: ${colors.shale};
+  }
+
+  &:hover {
+    background-color: ${colors.azure.base};
+    color: ${colors.white.base};
+  }
+
+  &:not(:last-of-type) {
+    margin-bottom: ${spacing.slim};
+  }
+`;
+
+const LinkItemContainer = ItemContainer.withComponent("a").extend`
+text-decoration: none;
+`;
+
+const ListItem = ({ children, index, onClick, isLink, ...rest }) => (
+  <LinkListConsumer>
```

Comment:
We can do that too but we will have to use React.cloneElement since we are passing ListItems as children props to List. Doing it this way to avoid the clone element and use renderProps instead